packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -150,26 +168,50 @@ class HorizontalOverflow extends Foundation<
     }
 
     /**
-     * Callback on scroll change
+     * Callback for on scroll change
      */
     private onScrollChange = (): void => {
-        if (!this.props.onScrollChange) {
-            return;
+        if (typeof this.props.onScrollChange === "function") {
+            this.props.onScrollChange(this.getScrollChangeData());
         }
+    };
 
+    /**
+     * Observe resize of parent element
+     */
+    private onResizeObserver = (): void => {
+        // TODO #1142 https://github.com/Microsoft/fast-dna/issues/1142
+        // Full browser support imminent
+        // Revisit usage once Safari and Firefox adapt
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1272409
+        // https://bugs.webkit.org/show_bug.cgi?id=157743
+        if (ResizeObserver) {
```

Comment:
how is this `if` check working? isn't `ResizeObserver` just a type definition of a class?

Reply:
@janechu I realized after removing this check, the solution you supplied was not actually working. The typings allow it to build, but the console outputs a type error `TypeError: _resize_observer__WEBPACK_IMPORTED_MODULE_7__.ResizeObserver is not a constructor`
I've been digging trying to figure out how to solve this, but I feel like I've hit a wall.

Any guidance from you or @nicholasrice would be helpful

Reply:
yea - thats cause the `ResizeObserver` you're importing is just a typings for the DOM resize observer.

You would check for it's existence with `window.ResizeObserver`.

Reply:
Right, but even to add that check I need to add it to the global namespace, because `window.ResizeObserver` is unknown.

The issue with the typings still remains and I get the same error. I am still unsure how to solve.

Reply:
You can try extending the window type and add ResizeObserver to it the explicitly try (window as WindowWithResizeObserver).ResizeObserver

Reply:
Should all be resolved now thanks @janechu